main/src/ca/uwaterloo/flix/language/phase/Monomorpher.scala

```diff
@@ -590,9 +600,9 @@ object Monomorpher {
               val env1 = env0 + (sym -> freshSym)
               val subst1 = caseSubst @@ subst.nonStrict
```

Comment:
It's hard to know if this is now doing the same thing, but I think its fine

Reply:
@mlutze Why do we need this? What's the consequence if it is made strict?

Reply:
If you typematch with the pattern `List[_]` and make that substitution strict, then you effectively match for `List[AnyType]` which ruins the pattern

Reply:
What's the example? (i.e. something that be a test case that breaks)

Reply:
1. I remember some discussion about List[_]. What was it?
2. Is my solution broken or not? I remain the same dom(s) for the two substitutions, so if x \notin dom(s) then x \notin dom(s') where s' is a subst. with my new implementation.